mlflow/server/js/src/model-registry/components/ModelVersionView.js

```diff
@@ -316,8 +317,18 @@ export class ModelVersionViewImpl extends React.Component {
         </a>
       );
     } else if (runInfo) {
+      if (!modelSource) {
+        throw new Error('model source is missing.');
+      }
+      const artifactPath = modelSource.match(/.+\/artifacts\/(.+)$/)[1];
```

Comment:
This line seems to assume that `modelSource` always ends with `/artifacts/...`. What if it doesn't?

Reply:
What's the cases it does not ends with /artifacts/... ? 

I think the source string is
```
[file|s3|...]:///path/to/mlruns/{experiment_id}/{run_id}/artifacts/{artifact_path}
```
or
```
mlflow-artifacts:/{experiment_id}/{run_id}/artifacts/{artifact_path}
```